### PR TITLE
Feature/#1240 medusa pull secret ref

### LINKED ISSUE #1240: The PullSecretRef for medusa is ignored in the standalone deployment of medusa
**What happened?**

The PullSecretRef for medusa is not set for the standalone medusa deployment. This leads that the helm operation fails, when the image comes from an internal docker registry.

I looked into the code and the configuration for the pullSecretRef is completely ignored for the medusa standalone deployment.

https://github.com/k8ssandra/k8ssandra-operator/blob/main/pkg/medusa/reconcile.go#L529

**Did you expect to see something different?**

That the configured pullSecretRef is  set as ImagePullSecrets at the deployment.

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**

* K8ssandra Operator version:

    `v1.13.0`

### PR DESCRIPTION
**Pull request by tuxBurner**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #1240 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated not needed IMHO because it fixes something what is already documentated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by adejanovski — APPROVED** [review]
Looks good, thanks for fixing this 👍

**Comment by adejanovski** [comment]
Hi @tuxBurner, your branches has some conflicts and can't make it into the release we're cutting out today.
Once the release is done, please fix those conflicts after rebasing and we'll issue a patch release with it.
Thanks!

**Comment by tuxBurner** [comment]
@adejanovski okay merge conflict is fixxed

**Comment by adejanovski** [comment]
> @adejanovski well now suddenly i get a failure on the Quality Gate check. But the Code was fine before 👍 Am i supposed to fix this or can I ignore it ?

You can ignore it, no worries

File: CHANGELOG/CHANGELOG-1.14.md [hunk 1]
```diff
@@ -23,3 +24,4 @@ When cutting a new release, update the `unreleased` heading to the tag being gen
 * [CHANGE] Update Medusa to v0.20.1
 * [BUGFIX] [#1235](https://github.com/k8ssandra/k8ssandra-operator/issues/1235) Prevent operator from modifying the spec when superUserRef is not set. Also, remove the injection to mount secrets to cassandra container.
 * [BUGFIX] [#1239](https://github.com/k8ssandra/k8ssandra-operator/issues/1239) Use `concurrent_transfers` setting from MedusaConfiguration object if it is actually set.
+* [BUGFIX] [#1240](https://github.com/k8ssandra/k8ssandra-operator/issues/1240) The PullSecretRef for medusa is ignored in the standalone deployment of medusa
```
Comment (adejanovski) [inline CHANGELOG/CHANGELOG-1.14.md:27]: **Issue:** 1.14.0 was released already and this line needs to go under a new `## unreleased` section to create above.

**Review by adejanovski — CHANGES_REQUESTED** [review]

File: CHANGELOG/CHANGELOG-1.14.md (hunk 1 above)
Comment (tuxBurner) [reply CHANGELOG/CHANGELOG-1.14.md]: Okay fixxed :)

**Review by adejanovski — APPROVED** [review]

**Comment by tuxBurner** [comment]
:partying_face: thanks a lot

**Comment by bradfordcp** [comment]
Thank you for your contribution @tuxBurner! 🎉

**Pull request merged**